src/main/java/nl/mpcjanssen/simpletask/Simpletask.kt

```diff
@@ -743,45 +738,15 @@ class Simpletask : ThemedActivity() {
                 val shareText = selectedTasksAsString()
                 shareText(this@Simpletask, "Simpletask tasks", shareText)
             }
-            R.id.context_archive -> archiveTasks(TodoList.selectedTasks, true)
-            R.id.context_calendar -> {
-                var calendarTitle = getString(R.string.calendar_title)
-                var calendarDescription = ""
-                if (checkedTasks.size == 1) {
-                    // Set the task as title
-                    calendarTitle = checkedTasks[0].task.text
-                } else {
-                    // Set the tasks as description
-                    calendarDescription = selectedTasksAsString()
-
-                }
-                intent = Intent(Intent.ACTION_EDIT).setType(Constants.ANDROID_EVENT).putExtra(Events.TITLE, calendarTitle).putExtra(Events.DESCRIPTION, calendarDescription)
-                // Explicitly set start and end date/time.
-                // Some calendar providers need this.
-                val dueDate =  checkedTasks[0].task.dueDate
-                val calDate = if (checkedTasks.size == 1 && dueDate != null ) {
-                    val year = dueDate.substring(0,4).toInt()
-                    val month =  dueDate.substring(5,7).toInt()-1
-                    val day =  dueDate.substring(8,10).toInt()
-                    GregorianCalendar(year, month, day)
-                } else {
-                    GregorianCalendar()
-                }
-
-                intent.putExtra(CalendarContract.EXTRA_EVENT_BEGIN_TIME,
-                        calDate.timeInMillis)
-                intent.putExtra(CalendarContract.EXTRA_EVENT_END_TIME,
-                        calDate.timeInMillis + 60 * 60 * 1000)
-                startActivity(intent)
-            }
+            R.id.context_archive -> archiveTasks(checkedTasks, true)
+            R.id.context_calendar -> createCalendarAppointment(checkedTasks)
```

Comment:
There's a *ton* of code here and it makes the list a lot harder to skim. So, let's move it into a helper function.